src/common/addlayers/AddLayersDirective.js

```diff
@@ -242,6 +219,13 @@
               scope.cart = [];
             };
 
+            scope.clearLayerFromCart = function(layerConfig) {
+              var configIndex = scope.cart.indexOf(layerConfig);
```

Comment:
Please try to reuse the code that is already there.
maybe refactor the existing code in `addToCart`